ui/handler.go

```diff
@@ -22,21 +22,30 @@ var (
 )
 
 // NewHandler creates a http handler serving the UI application and the UI backend.
-func NewHandler(service store.RulesetService, distPath string) http.Handler {
+func NewHandler(service store.RulesetService, fs http.FileSystem) http.Handler {
 	var mux http.ServeMux
 
 	// internal API
 	mux.Handle("/i/", http.StripPrefix("/i", newInternalHandler(service)))
 
 	// static files
-	fs := http.FileServer(http.Dir(distPath))
-	mux.Handle("/css/", fs)
-	mux.Handle("/js/", fs)
-	mux.Handle("/fonts/", fs)
+	h := http.FileServer(fs)
+	mux.Handle("/css/", h)
+	mux.Handle("/js/", h)
+	mux.Handle("/fonts/", h)
 
 	// catch all url that deleguates the routing to the front app router
 	mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
-		http.ServeFile(w, r, filepath.Join(distPath, "index.html"))
+		f, err := fs.Open("index.html")
+		if err != nil {
+			writeError(w, r, err, http.StatusInternalServerError)
+			return
+		}
```

Comment:
The file must be closed once you're done with it

```suggestion
		}
		defer f.Close()
```